src/main/java/com/synopsys/integration/alert/web/security/authentication/ldap/LdapManager.java

```diff
@@ -38,64 +42,80 @@
-    private LdapProperties currentConfiguration;
+    private final ConfigurationAccessor configurationAccessor;
 
     @Autowired
-    public LdapManager(final LdapProperties ldapProperties) {
-        this.currentConfiguration = ldapProperties;
+    public LdapManager(final ConfigurationAccessor configurationAccessor) {
+        this.configurationAccessor = configurationAccessor;
     }
 
     public boolean isLdapEnabled() {
-        return currentConfiguration != null && currentConfiguration.isEnabled();
-    }
+        boolean enabled = false;
+        try {
+            enabled = Boolean.valueOf(getFieldValueOrEmpty(getCurrentConfiguration(), SettingsDescriptor.KEY_LDAP_ENABLED));
+        } catch (final AlertDatabaseConstraintException | AlertLDAPConfigurationException ex) {
+            logger.error("Error checking if Ldap is enabled", ex);
+        }
 
-    public LdapProperties getCurrentConfiguration() {
-        return currentConfiguration;
+        return enabled;
     }
 
-    public LdapAuthenticationProvider getAuthenticationProvider() throws AlertLDAPConfigurationException {
-        if (authenticationProvider == null) {
-            updateConfiguration(currentConfiguration);
+    public ConfigurationModel getCurrentConfiguration() throws AlertDatabaseConstraintException, AlertLDAPConfigurationException {
+        final List<ConfigurationModel> configurations = configurationAccessor.getConfigurationsByDescriptorName(SettingsDescriptor.SETTINGS_COMPONENT);
+        if (!configurations.isEmpty() && 1 == configurations.size()) {
+            return configurations.get(0);
         }
-        return authenticationProvider;
+        throw new AlertLDAPConfigurationException("more than 1 settings configuration present");
     }
 
-    public void updateConfiguration(final LdapProperties configuration) throws AlertLDAPConfigurationException {
-        this.currentConfiguration = configuration;
-        if (currentConfiguration.isEnabled()) {
-            updateContext();
-        }
+    public LdapAuthenticationProvider getAuthenticationProvider() throws AlertLDAPConfigurationException {
+        updateContext();
+        return authenticationProvider;
     }
 
     public void updateContext() throws AlertLDAPConfigurationException {
-        final LdapProperties configuration = getCurrentConfiguration();
-        final LdapContextSource ldapContextSource = new LdapContextSource();
-
         try {
-            if (StringUtils.isNotBlank(configuration.getServer())) {
-                ldapContextSource.setUrl(configuration.getServer());
-                ldapContextSource.setUserDn(configuration.getManagerDN());
-                ldapContextSource.setPassword(configuration.getManagerPassword());
-                ldapContextSource.setReferral(configuration.getLdapReferral());
-                ldapContextSource.setAuthenticationStrategy(createAuthenticationStrategy(configuration));
+            if (isLdapEnabled()) {
```

Comment:
We might want to consider making this short-circuit since nothing will get executed if this  returns false. I don't know if we are trying to minimize the number of return paths, but nesting a lot of code can cause readability issues when unnecessary.